@here/harp-mapview/lib/geometry/PhasedTileGeometryManager.ts

```diff
@@ -56,6 +56,12 @@ export class PhasedTileGeometryManager extends TileGeometryManagerBase {
             ? this.updateSomeTiles(tiles)
             : this.updateAllTilesTogether(tiles);
 
+        if (this.m_tileUpdateCallback) {
+            for (const tile of tiles) {
+                this.m_tileUpdateCallback(tile);
```

Comment:
is this ok to call this on a tile which wasn't updated. @StefanDachwitz , can you comment?

Reply:
I mean to say, above, based on the isDynamicFrame flag, you could have just updated some tiles, and not all, but here you call the callback on all tiles....

Reply:
that's correct. i need to call the overlay function for all tiles anyway, even if they weren't updated, for two reasons:
1. The elevation data may have changed because the final terrain tile is now loaded.
2. The text overlay will continue after the tile is loaded, since it's spread to multiple frames.

So right now the callback is called for any tile that goes through the TileGeometryManager.updateTiles() method.

I agree it's confusing though. I'll try to think on how to improve this.